### PR TITLE
Added support for POSIX line endings at end of file - Closes #798

### LINKED ISSUE #798: Number of lines returned by read_file() with negative starting point is 1 fewer than expected
It looks like the starting index might be off by 1.

**Test file:**
```
> more TEST_READ_FILE
::::::::::::::
/u/g/gesslar/TEST_READ_FILE
::::::::::::::

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
```

```
> eval return read_file("/u/g/gesslar/TEST_READ_FILE", -1)
Result = ""
```

```
> eval return sizeof( explode(read_file("/u/g/gesslar/TEST_READ_FILE", -5), "\n") )
Result = 4
```

### PR DESCRIPTION
**Pull request by jalbright015**
Due to different line ending standards (POSIX vs non-POSIX), support was added in to handle both of these.

### CODE REVIEW

File: testsuite/single/tests/efuns/read_file.c [hunk 1]
```diff
@@ -20,11 +20,9 @@ void do_tests() {
...
     
     // Start tests for end of file -lines
-    mid = implode(explode(foo, "\n")[<14..<5], "\n") + "\n";
-    ASSERT_EQ(mid, read_file("/testfile", -15, 10));
+    ASSERT_EQ(implode(explode(foo, "\n")[<15..<6], "\n") + "\n", read_file("/testfile", -15, 10));
     // if we start too many lines before the end of the file, start at the beginning
     ASSERT(read_file("/testfile", -10000, 1));
-
     // CRLF tests
     ASSERT(read_file("/crlf_test_file.lf") == read_file("/crlf_test_file.crlf"));
```
Comment (thefallentree) [inline testsuite/single/tests/efuns/read_file.c:27]: Can you add a test case for negative reading using test file.crlf ?

Comment (thefallentree) [reply testsuite/single/tests/efuns/read_file.c:29]: Well this is only testing out-of-range cases, I think we need to cover normal cases as well, the same case as above.

It should work, but i think we might need a test.

File: testsuite/single/tests/efuns/read_file.c [hunk 2]
```diff
@@ -20,12 +20,12 @@ void do_tests() {
...
+    ASSERT_EQ(implode(explode(foo, "\n")[<15..<6], "\n") + "\n", read_file("/testfile", -15, 10));
     // if we start too many lines before the end of the file, start at the beginning
     ASSERT(read_file("/testfile", -10000, 1));
-
+    
+    
     // CRLF tests
     ASSERT(read_file("/crlf_test_file.lf") == read_file("/crlf_test_file.crlf"));
-
+    ASSERT(read_file("/crlf_test_file.lf", -1000) == read_file("/crlf_test_file.crlf", -1000));
```
Comment (thefallentree) [inline testsuite/single/tests/efuns/read_file.c:30]:
```suggestion
    ASSERT_EQ(read_file("/crlf_test_file.lf", -2, 1), read_file("/crlf_test_file.crlf", -2, 1));
    ASSERT_EQ(read_file("/crlf_test_file.lf", -1000), read_file("/crlf_test_file.crlf", -1000));
```
